### ISSUE TITLE
[Llama] Make Llama in torchao trainable

### ISSUE THREAD

**Issue by gau-nernst** [pytorch/ao#674]
While working on #644, @msaroufim suggested to use the built-in Llama for testing the mini train recipe. I looked into it and here are the 2 main changes to be made.

1. Initialize `freq_cis` without initializing KV-Cache and causal mask https://github.com/pytorch/ao/blob/e7fc0ed0f2e130e44d250541aba71c7a7a985f15/torchao/_models/llama/model.py#L153-L170
2. Don't use attention mask, just use `is_causal=True` directly https://github.com/pytorch/ao/blob/e7fc0ed0f2e130e44d250541aba71c7a7a985f15/torchao/_models/llama/model.py#L247

This will make it convenient for some of our training recipes (e.g. QAT) to have a mini training scripts directly in torchao, and also act as self-contained examples.

API wise, I think we can add a `training` flag to `Transformer.setup_caches()` method.
- When `training=False` (default), the old behavior is maintained.
- When `training=True`, only `freq_cis` is initialized, and in the `.forward()` method, we don't pass `mask` to TransformerBlock/Attention.
